File: spring-microservices/course-service/src/main/java/com/hoangphihiep/dto/response/PublicCertificateVerificationResponse.java
package com.hoangphihiep.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class PublicCertificateVerificationResponse {
    private boolean found;
    private CertificateResponse certificate;

    private boolean onChainChecked;
    private Boolean onChainValid;
    private Boolean dataMatched;

    private String onChainUserId;
    private Integer onChainPublishedCourseId;
    private Date onChainIssueDate;

    private String message;
}

File: react-type-vite/src/services/api/response/publicCertificateVerificationResponse.ts
import type { CertificateResponse } from "./certificateResponse";

export interface PublicCertificateVerificationResponse {
  found: boolean;
  certificate: CertificateResponse | null;

  onChainChecked: boolean;
  onChainValid: boolean | null;
  dataMatched: boolean | null;

  onChainUserId: string | null;
  onChainPublishedCourseId: number | null;
  onChainIssueDate: string | null;

  message: string;
}
